src/components/SVGImage.jsx

```diff
@@ -0,0 +1,69 @@
+import React, { PropTypes } from 'react';
+
+export default class SVGImage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loaded: false,
+      error: false,
+    };
+    
+    this.image = new Image();
+    this.image.onload = () => {
+      if (this.props.onLoad) this.props.onLoad(this.image);
+      this.setState({
+        loaded: true,
+      });
+    };
+    this.image.onerror = (err) => {
+      if (this.props.onError) this.props.onError(err);
+      this.setState({
+        error: true,
+      });
+    };
+    
+    if (this.props.src) {
+      this.image.src = this.props.src;
+    } else {
+      this.state.loaded = false;
+      this.state.error = true;
+    }
+  }
+  
+  render() {
+    if (this.state.loaded) {
+      return (
+        <image className="svg-image"
+          href={this.image.src}
+          width={this.image.width}
+          height={this.image.height}
+          x={(this.image.width * -0.5)+'px'}
+          y={(this.image.height * -0.5)+'px'} />
```

Comment:
Yup, it's a centre-ing compensator. Might be worth updating this with some comments to explain that.